src/softmax.jl

```diff
@@ -61,23 +74,25 @@ It is semantically equivalent to the following:
 See also [`softmax`](@ref).
 """
 logsoftmax(x; dims = 1) = logsoftmax!(similar(x, (float ∘ eltype)(x)), x; dims = dims)
+
 logsoftmax!(x; dims = 1) = logsoftmax!(x, x; dims = dims)
+
 function logsoftmax!(out::O, x::T; dims = 1) where {O<:AbstractArray,T<:AbstractArray}
     out .= x .- maximum(x; dims = dims)
     # out .-= log.(sum(exp.(out); dims = dims))  # WARN: this will decrease performance.
     log_ = log.(sum(exp.(out); dims = dims))
     out .-= log_
```

Comment:
Maybe you can save a few allocations here?
```suggestion
    tmp_ = sum(exp, out; dims = dims)
    tmp_ .= log.(tmp_)
    out .-= tmp_
```